tests/display/test_auto_config.py

```diff
@@ -0,0 +1,57 @@
+# bluemira is an integrated inter-disciplinary design tool for future fusion
+# reactors. It incorporates several modules, some of which rely on other
+# codes, to carry out a range of typical conceptual fusion reactor design
+# activities.
+#
+# Copyright (C) 2021 M. Coleman, J. Cook, F. Franza, I.A. Maione, S. McIntosh, J. Morris,
+#                    D. Short
+#
+# bluemira is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# bluemira is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with bluemira; if not, see <https://www.gnu.org/licenses/>.
+
+"""
+Tests for the display auto_config module.
+
+"""
+import time
+from functools import partial
+from unittest.mock import patch
+
+from bluemira.display.auto_config import get_primary_screen_size
+
+
+def timeout(t=4):
+    """
+    The timeout in get_primary_screen_size is 3s
+    """
+    time.sleep(t)
+    return True
+
+
+class TestGetScreenSize:
+    def setup(self):
+        get_primary_screen_size.cache_clear()
+
+    @patch("bluemira.display.auto_config._get_primary_screen_size", new=timeout)
+    def test_timeout(self, caplog):
```

Comment:
Having these 2 tests take 4 seconds in total feels unnecessary. Could we make the timeout duration an argument to the function and reduce it to something like 0.05s in the tests?